pkg/algorithm/hived_algorithm.go

```diff
@@ -687,72 +690,93 @@ func (h *HivedAlgorithm) scheduleNewAffinityGroup(
+	} else if physicalPlacement != nil {
+		klog.Infof("Found placement for group %v but not fully in within suggested nodes", s.AffinityGroup.Name)
 	} else {
-		klog.Infof("Failed to schedule group %v", s.AffinityGroup.Name)
+		klog.Infof("Failed to find placement for group %v", s.AffinityGroup.Name)
 	}
-	return physicalPlacement, virtualPlacement
+	return physicalPlacement, virtualPlacement, nodesNotInSuggested
 }
 
-// scheduleAffinityGroupForGivenGpuType schedules an affinity group in a certain cell chain
-// that matches the specified GPU type.
-func (h *HivedAlgorithm) scheduleAffinityGroupForGivenGpuType(
+// scheduleAffinityGroupForGpuType schedules an affinity group in a certain cell chain
+// that matches the given GPU type.
+func (h *HivedAlgorithm) scheduleAffinityGroupForGpuType(
 	sr schedulingRequest,
 	gpuType string,
 	pod *core.Pod,
-	suggestedNodes common.Set) (groupPhysicalPlacement, groupVirtualPlacement) {
+	suggestedNodes common.Set,
+	typeSpecified bool) (
+	physicalPlacement groupPhysicalPlacement,
+	virtualPlacement groupVirtualPlacement,
+	nodesNotInSuggested common.Set) {
 
-	if chains := h.chains[gpuType]; chains == nil {
-		panic(internal.NewBadRequestError(fmt.Sprintf(
-			"[%v]: Pod requesting GPU type %v which the whole cluster does not have",
-			internal.Key(pod), gpuType)))
-	} else {
-		vcHasType := false
-		for _, chain := range chains {
-			if h.vcSchedulers[sr.vc].getNonReservedFullCellList()[chain] != nil {
-				vcHasType = true
-				sr.chain = chain
-				physicalPlacement, virtualPlacement := h.processSchedulingRequest(sr, suggestedNodes)
-				if physicalPlacement != nil {
-					return physicalPlacement, virtualPlacement
-				}
-			}
+	vcHasType := false
+	for _, chain := range h.chains[gpuType] {
+		if h.vcSchedulers[sr.vc].getNonReservedFullCellList()[chain] != nil {
+			vcHasType = true
 		}
-		if sr.priority >= minGuaranteedPriority && !vcHasType {
-			panic(internal.NewBadRequestError(fmt.Sprintf(
-				"[%v]: Pod requesting GPU type %v which VC %v does not have",
-				internal.Key(pod), gpuType, sr.vc)))
+		klog.Infof("Searching chain %v", chain)
+		sr.chain = chain
+		physicalPlacement, virtualPlacement = h.processSchedulingRequest(sr, suggestedNodes)
+		nodesNotInSuggested = collectNodesNotSuggested(physicalPlacement, suggestedNodes)
+		if physicalPlacement != nil && nodesNotInSuggested.IsEmpty() {
+			klog.Infof("Successfully found placement in chain %v fully within suggested nodes: %v",
+				chain, physicalPlacement)
+			return physicalPlacement, virtualPlacement, nodesNotInSuggested
+		} else if physicalPlacement != nil {
+			klog.Infof("Found placement in chain %v but not fully within suggested nodes: "+
+				"%v, non-suggested nodes: %v. Continue to search other chains",
+				chain, physicalPlacement, nodesNotInSuggested)
+		} else {
+			klog.Infof("No placement found in chain %v", chain)
 		}
 	}
-	return nil, nil
+	if typeSpecified && sr.priority >= minGuaranteedPriority && !vcHasType {
```

Comment:
>sr.priority >= minGuaranteedPriority [](start = 21, length = 36)

Still skip chains that the VC does not have if it is not oppo?

Reply:
this logic is moved to processSchedulingRequest